File: zb1/codingTest/220317/RemoveDuplicatedWord/RemoveDuplicatedWord.js
/**
 * @param s {string}
 * @returns {string}
 */
function solution(s) {
  const stack = [];
  for (const c of s) {
    if (stack.length && stack[stack.length - 1] === c) {
      stack.pop();
    } else {
      stack.push(c);
    }
  }
  var answer = "";
  for (const c of stack) {
    answer = answer.concat(c);
  }
  return answer;

}

solution

File: zb1/codingTest/220317/RemoveDuplicatedWord/RemoveDuplicatedWord.py
def solution(s):
    """
    :param s: str
    :return: str
    """

    stack = []
    for c in s:
        if len(stack) > 0 and stack[-1] == c:
            stack.pop()
        else:
            stack.append(c)

    result = ''
    for c in stack:
        result += c

    return result
